### PR TITLE
Add kubernetes example deployment

### PR DESCRIPTION
**Pull request by tico24**
I appreciate that it's not a huge amount of work, but I deployed Sorry Cypress into Kubernetes and I thought this might help others in the future.

### CODE REVIEW

File: kubernetes-full.yml [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+          serviceName: director-service
+          servicePort: 1234
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cypress-s3-secrets
+data:
+  AWS_SECRET_ACCESS_KEY: VGhpcyBpcyB0aGUgY3J1bWJob2xl
+  AWS_ACCESS_KEY_ID: Tm8gcGVla2luZw==
```
Comment (agoldis) [inline kubernetes-full.yml:293]: 👀

File: kubernetes-full.yml [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    metadata:
+      labels:
+        app: dashboard
+      name: dashboard
+    spec:
+      containers:
+      - env:
+        - name: GRAPHQL_SCHEMA_URL
+          value: [YOUR-API-URL-HERE]
+        image: agoldis/sorry-cypress-dashboard:cypress-v5
```
Comment (agoldis) [inline kubernetes-full.yml:190]: I have just merged cypress v5 into master
```suggestion
        image: agoldis/sorry-cypress-dashboard:latest
```

File: kubernetes-full.yml [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+        app: api
+      name: api
+    spec:
+      containers:
+      - env:
+        - name: MONGODB_DATABASE
+          value: sorry-cypress
+        - name: MONGODB_URI
+          value: mongodb://mongo-service:27017
+        image: agoldis/sorry-cypress-api:cypress-v5
```
Comment (agoldis) [inline kubernetes-full.yml:82]:
```suggestion
        image: agoldis/sorry-cypress-api:latest
```

File: kubernetes-full.yml [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+          valueFrom:
+              secretKeyRef:
+                name: cypress-s3-secrets
+                key: AWS_ACCESS_KEY_ID
+        - name: AWS_SECRET_ACCESS_KEY
+          valueFrom:
+              secretKeyRef:
+                name: cypress-s3-secrets
+                key: AWS_SECRET_ACCESS_KEY
+        image: agoldis/sorry-cypress-director:cypress-v5
```
Comment (agoldis) [inline kubernetes-full.yml:147]:
```suggestion
        image: agoldis/sorry-cypress-director:latest
```

File: kubernetes-full.yml (hunk 1 above)
Comment (tico24) [reply kubernetes-full.yml:293]: Placeholders, don't worry :)

**Comment by agoldis** [comment]
That's awesome! Why not huge, it's very helpful, i think it will be quite an enabler.
Any chance you write few paragraphs of instruction - I can use it for documentation and for myselft

**Comment by tico24** [comment]
I am half-way through writing a blog if it helps?
https://crumbhole.com/playing-with-sorry-cypress-and-kubernetes/

File: kubernetes-full.yml [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+      - backend:
+          serviceName: dashboard-service
+          servicePort: 8080
+  - host: [YOUR-API-URL-HERE]
+    http:
+      paths:
+      - backend:
+          serviceName: api-service
+          servicePort: 4000
+  - host: [YOUR-DIRECTOR-URL-HERE]
```
Comment (agoldis) [inline kubernetes-full.yml:280]: Sorry if I am stupid, does it allow to access "internal" services using the URLs in `[]`?

File: kubernetes-full.yml [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+  template:
+    metadata:
+      labels:
+        app: dashboard
+      name: dashboard
+    spec:
+      containers:
+      - env:
+        - name: GRAPHQL_SCHEMA_URL
+          value: [YOUR-API-URL-HERE]
```
Comment (agoldis) [inline kubernetes-full.yml:189]: Is it possible to use some kind of "internal DNS" to route traffic to API service? I have 0 kubernetes experience

File: kubernetes-full.yml (hunk 5 above)
Comment (tico24) [reply kubernetes-full.yml:280]: This is essentially what creates the url, provisioning a loadbalancer and even possibly the DNS entries (if you have k8s set up to do it).

**Comment by agoldis** [comment]
I will run this soon, right in the middle of a release - awesome work ! 🎉

File: kubernetes-full.yml (hunk 6 above)
Comment (tico24) [reply kubernetes-full.yml:189]: Yes, that's possible. I actually already do that on line 130. The director pod accesses mongo internally.
If that's what's supposed to happen here then I'll adjust.

**Comment by tico24** [comment]
Actually, I've just spotted a problem with the api readiness probe. Hold on and I'll push a revised yaml.

File: kubernetes-full.yml (hunk 6 above)
Comment (agoldis) [reply kubernetes-full.yml:189]: Yes, `dashboard` pull graphql schema definition from the API service

Comment (tico24) [reply kubernetes-full.yml:189]: So it looks like the dashboard also relies on this URL when the end user visits the dashboad through a browser.
Changing this to the internally-accessible dns entry prevents the end user from loading their data.

So I believe the right thing to do in this instance is indeed to call the external API URL.

Comment (agoldis) [reply kubernetes-full.yml:181]: Oh right, it need to access the API from the user-space

**Comment by agoldis** [comment]
@all-contributors please add @tico24 for Content and example

**Comment by agoldis** [comment]
@all-contributors please add @tico24 for content and example

**Review by agoldis — APPROVED** [review]

**Pull request merged**